src/components/Toaster/Toast/Toast.tsx

```diff
@@ -0,0 +1,267 @@
+import {AttentionToast} from '../../icons/AttentionToast';
+import {SuccessToast} from '../../icons/SuccessToast';
+
+import './Toast.scss';
+
+const b = block('toast');
+
+const FADE_IN_LAST_ANIMATION_NAME = 'move-left';
+const FADE_OUT_LAST_ANIMATION_NAME = 'remove-height';
+
+const DEFAULT_TIMEOUT = 5000;
+
+const TITLE_ICONS = {
+    error: AttentionToast,
+    success: SuccessToast,
+};
+
+export interface ToastAction {
+    label: string;
+    onClick: VoidFunction;
+    removeAfterClick?: boolean;
+}
+
+export type ToastType = 'error' | 'success';
+
+export interface ToastGeneralProps {
+    name: string;
+    title?: string;
+    className?: string;
+    timeout?: number;
+    allowAutoHiding?: boolean;
+    content?: React.ReactNode;
+    type?: ToastType;
+    isClosable?: boolean;
+    isOverride?: boolean;
+    actions?: ToastAction[];
+}
+
+interface ToastInnerProps {
+    removeCallback: VoidFunction;
+}
+
+interface ToastProps extends ToastGeneralProps, ToastInnerProps {}
+
+enum ToastStatus {
+    creating = 'creating',
+    showingIndents = 'showing-indents',
+    showingHeight = 'showing-height',
+    hiding = 'hiding',
+    shown = 'shown',
+}
+
+type ToastStyles = {
+    height?: number;
+    position?: 'relative';
+};
+
+interface UseCloseOnTimeoutProps {
+    onClose: VoidFunction;
+    timeout?: number;
+}
+
+function useCloseOnTimeout({onClose, timeout}: UseCloseOnTimeoutProps) {
+    const timerId = React.useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
+
+    const setTimer = React.useCallback(() => {
+        if (!timeout) {
+            return;
+        }
+
+        timerId.current = setTimeout(async () => {
+            onClose();
+        }, timeout);
+    }, [timeout, onClose]);
+
+    const clearTimer = React.useCallback(() => {
+        if (timerId.current) {
+            clearTimeout(timerId.current);
+            timerId.current = undefined;
+        }
+    }, []);
+
+    React.useEffect(() => {
+        setTimer();
+        return () => {
+            clearTimer();
+        };
+    }, [setTimer, clearTimer]);
+
+    const onMouseOver = () => {
+        clearTimer();
+    };
+
+    const onMouseLeave = () => {
+        setTimer();
+    };
+
+    return {onMouseOver, onMouseLeave};
+}
+
+interface UseHeightProps {
+    isOverride: boolean;
+    status: ToastStatus;
+}
+
+function useToastHeight({isOverride, status}: UseHeightProps) {
+    const [height, setHeight] = React.useState<number | undefined>(undefined);
+
+    const ref = React.useRef<HTMLDivElement>(null);
+    const style: ToastStyles = {};
+    const getToastHeight = React.useCallback(() => {
+        return ref.current?.offsetHeight;
+    }, [ref]);
+
+    React.useEffect(() => {
+        setHeight(getToastHeight());
+    }, [getToastHeight]);
+
+    React.useEffect(() => {
+        if (isOverride) {
+            setHeight(getToastHeight());
+        }
+    }, [isOverride, getToastHeight]);
```

Comment:
This doesn't  work if override twice with different content. 

```suggestion
    React.useEffect(() => {
        if (isOverride) {
            setHeight(getToastHeight());
        }
    });
```